samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/PublishedAssessmentSettingsBean.java

```diff
@@ -411,18 +411,8 @@ public void setAssessment(PublishedAssessmentFacade assessment) {
       setIpAddresses(assessment);
 
       // publishedUrl
-      FacesContext context = FacesContext.getCurrentInstance();
-      ExternalContext extContext = context.getExternalContext();
-      // get the alias to the pub assessment
-      this.alias = assessment.getAssessmentMetaDataByLabel(
-          AssessmentMetaDataIfc.ALIAS);
-      String server = ( (javax.servlet.http.HttpServletRequest) extContext.
-                       getRequest()).getRequestURL().toString();
-      int index = server.indexOf(extContext.getRequestContextPath() + "/"); // "/samigo-app/"
-      server = server.substring(0, index);
-      String url = server + extContext.getRequestContextPath();
-      this.publishedUrl = url + "/servlet/Login?id=" + this.alias;
-      
+      generatePublishedURL(assessment);
```

Comment:
minor, but i find it easier to visualize what code is doing when 

```suggestion
      this.publishedUrl = generatePublishedURL(assessment);
```

Reply:
Currently the function doesn't return anything, I just copied/pasted the code to a new method so i wouldn't have to run `.setAssessment()` in entirety to get the URL generated.

Would you prefer the new method to return the URL rather than set it internally? I don't have a strong preference either way.

Reply:
i think it's more readable when it just returns a String .... but not a big deal

Reply:
No worries, I'll change it. Just requires a little more refactoring than your suggested patch. Incoming...